src/Apps/Order/Utils/getErrorDialogCopy.tsx

```diff
@@ -0,0 +1,80 @@
+import { RouterLink } from "System/Router/RouterLink"
+import React from "react"
+
+const SUPPORT_EMAIL = "orders@artsy.net"
+
+export enum ErrorDialogs {
+  CurrencyNotSupported = "currency_not_supported",
+}
+
+interface ErrorDialogCopy {
+  title: string
+  message: string
+  formattedMessage: React.ReactNode
+}
+
+export const getErrorDialogCopy = (dialog?: ErrorDialogs): ErrorDialogCopy => {
+  switch (dialog) {
+    case ErrorDialogs.CurrencyNotSupported:
+      return currencyNotSupportedErrorDialogCopy()
+    default:
+      return defaultErrorDialogCopy()
+  }
+}
+
+const currencyNotSupportedErrorDialogCopy = () => {
+  const title = "Payment declined"
+  const message = `This card is not compatible with the currency for this artwork. Please confirm with your card issuer if this currency can be supported, try a different payment method or contact ${SUPPORT_EMAIL}.`
+
+  return errorDialogCopy(title, message)
+}
+
+const defaultErrorDialogCopy = () => {
+  const title = "An error occurred"
+  const message = `Something went wrong. Please try again or contact ${SUPPORT_EMAIL}.`
+
+  return errorDialogCopy(title, message)
+}
+
+const errorDialogCopy = (title: string, message: string): ErrorDialogCopy => {
+  return {
+    title,
+    message,
+    formattedMessage: formatErrorDialogMessage(message),
+  }
+}
+
+export const formatErrorDialogMessage = (message: string) => {
+  return (
+    <div data-testid="formatted-message">
+      {replaceSupportEmailWithMailtoLinks(message)}
+    </div>
+  )
+}
+
+const replaceSupportEmailWithMailtoLinks = (message: string) => {
```

Comment:
Ditto here -- component + props. When we start interpolating html as strings things can get a little bit confusing through time. Better to name things like one would a proper component tree:

```jsx
<EmailLink message={someMessage} />
```

Reply:
I moved this logic into the new `ErrorDialogMessage` component because I think that it's "setup code" rather than a separate component. (All it's doing is taking the split substrings and sandwiching mailto links between them.)

What do you think?

Reply:
I like 👌